test/test_python.py

```diff
@@ -20,14 +20,12 @@
 
 def run_greetings(port: int, user_num: int):
     try:
+        while socket.socket().connect_ex((CORE_ADDR, port)) == 0:
```

Comment:
If you allow the user to specify a port number, you should make sure that happens, or throw an error. 
Maybe we can only do random port num when it's not specified?

Reply:
I observed a rare case that the port number the test case specified was already occupied causing the test case to fail. Maybe I remove this port number check to outside?